presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -317,7 +317,7 @@ private void processNewSources()
                 }
 
                 // if current operator is finished...
-                if (current.isFinished()) {
+                if (current.isFinished() && !next.isFinished() && isBlocked(next).isDone()) {
```

Comment:
`isBlocked` is treated as only a suggestion from the operator and I don't think we should we should be changing that.

Both of these changes are something that the operator need to code for so I don't think we should be changing this.